src/resolve/resolvectl.c

```diff
@@ -2930,6 +2950,118 @@ static int verb_show_cache(int argc, char *argv[], void *userdata) {
+                { "failed-udp-attempts",        JSON_VARIANT_UNSIGNED,  json_dispatch_uint64,        offsetof(struct server_state, n_failed_udp),               JSON_MANDATORY },
+                { "failed-tcp-attempts",        JSON_VARIANT_UNSIGNED,  json_dispatch_uint64,        offsetof(struct server_state, n_failed_tcp),               JSON_MANDATORY },
+                { "seen-truncated-packet",      JSON_VARIANT_BOOLEAN,   json_dispatch_boolean,       offsetof(struct server_state, packet_truncated),           JSON_MANDATORY },
+                { "seen-opt-rr-getting-lost",   JSON_VARIANT_BOOLEAN,   json_dispatch_boolean,       offsetof(struct server_state, packet_bad_opt),             JSON_MANDATORY },
+                { "seen-rrsig-rr-missing",      JSON_VARIANT_BOOLEAN,   json_dispatch_boolean,       offsetof(struct server_state, packet_rrsig_missing),       JSON_MANDATORY },
+                { "seen-invalid-packet",        JSON_VARIANT_BOOLEAN,   json_dispatch_boolean,       offsetof(struct server_state, packet_invalid),             JSON_MANDATORY },
+                { "server-dropped-do-flag",     JSON_VARIANT_BOOLEAN,   json_dispatch_boolean,       offsetof(struct server_state, packet_do_off),              JSON_MANDATORY },
+                {},
+        };
+
+        r = json_dispatch(server, dispatch_table, NULL, JSON_LOG, &server_state);
+        if (r < 0)
+                return r;
+
+        printf("%sServer %s", ansi_underline(), server_state.server_name);
+        printf(" type=%s", server_state.type);
+
+        if (server_state.ifname)
+                printf(" interface=%s", server_state.ifname);
+
+        printf("%s\n", ansi_normal());
+
+        if (server_state.verified_feature_level)
+                printf("\tVerified feature level: %s\n", server_state.verified_feature_level);
+
+        if (server_state.possible_feature_level)
+                printf("\tPossible feature level: %s\n", server_state.possible_feature_level);
+
+        printf("\tDNSSEC Mode: %s\n", server_state.dnssec_mode);
+        printf("\tCan do DNSSEC: %s\n", yes_no(server_state.can_do_dnssec));
+        printf("\tMaximum UDP fragment size received: %zu\n", server_state.received_udp_fragment_max);
+        printf("\tFailed UDP attempts: %u\n", server_state.n_failed_udp);
+        printf("\tFailed TCP attempts: %u\n", server_state.n_failed_tcp);
+        printf("\tSeen truncated packet: %s\n", yes_no(server_state.packet_bad_opt));
+        printf("\tSeen OPT RR getting lost: %s\n", yes_no(server_state.packet_bad_opt));
+        printf("\tSeen RRSIG RR missing: %s\n", yes_no(server_state.packet_rrsig_missing));
+        printf("\tSeen invalid packet: %s\n", yes_no(server_state.packet_invalid));
+        printf("\tServer dropped DO flag: %s\n", yes_no(server_state.packet_do_off));
+        printf("\n");
```

Comment:
Please output this with the new table_new_vertical() logic. See how hostnamectl uses this.

Reply:
Updated